midp/gfx.js

```diff
@@ -112,8 +112,34 @@ var currentlyFocusedTextEditor;
         return 1;
     };
 
+    var hasNewFrame = true;
+    var ctxToExecuteOnNewFrame;
+    function gotNewFrame(timestamp) {
+        if (ctxToExecuteOnNewFrame) {
+            ctxToExecuteOnNewFrame.execute();
+            ctxToExecuteOnNewFrame = null;
+            window.requestAnimationFrame(gotNewFrame);
+        } else {
+            hasNewFrame = true;
+        }
+    }
 
-    addUnimplementedNative("com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V");
+    Native["com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V"] = function() {
+        // This is a no-op: The foreground display gets drawn directly
+        // to the screen.
+        //
+        // However, we sometimes "draw" more frequently than the
+        // screen can update. By calling `requestAnimationFrame` here,
+        // we ensure that whatever we just drew will stick around for
+        // one frame
+        if (hasNewFrame) {
+            hasNewFrame = false;
+            window.requestAnimationFrame(gotNewFrame);
+        } else {
+            ctxToExecuteOnNewFrame = $.ctx;
+            $.pause("Async");
```

Comment:
I copied this from our `AsyncImpl` implementation, so it seems like `AsyncImpl` has the same issue? Filed #1371. I'll fix both issues in this PR